activerecord/lib/active_record/attribute.rb

```diff
@@ -93,7 +93,7 @@ def came_from_user?
     end
 
     def has_been_read?
-      defined?(@value)
+      !defined?(@value).nil?
```

Comment:
But this is semantically diferent. Before it was checking if `@value` was defined, not is it checking if the return of `@value` is not `nil`. I think it is better to use `instance_variable_defined?`